tests/tron.py

```diff
@@ -197,20 +197,37 @@ def get_next_length(self, tx):
             return newpos
         return size + newpos
 
-    def navigate(self, snappath: Path = None, text: str = ""):
-        if self._firmware.device == "stax":
-            self._navigator.navigate_until_text_and_compare(
+    def navigate(self,
+                 snappath: Path = None,
+                 text: str = "",
+                 warning_approve: bool = False):
+        if not self._firmware.device.startswith("nano"):
+            path_name = ""
+            screen_change_before_first_instruction = True
+            if warning_approve:
                 # Use custom touch coordinates to account for warning approve
                 # button position.
-                NavIns(NavInsID.TOUCH, (200, 545)),
-                [
+                instructions = [
+                    NavIns(
+                        NavInsID.TOUCH,
+                        (200, 445 if self._firmware.device.startswith("flex")
+                         else 545)),
+                ]
+                self._navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH,
+                                                     str(snappath) + "/part1",
+                                                     instructions)
+                path_name = "/part2"
+                screen_change_before_first_instruction = False
+            self._navigator.navigate_until_text_and_compare(
```

Comment:
Should probably be replaced by `scenario_navigator`:
https://github.com/LedgerHQ/app-?ethereum/blob/76e8690a68fb84910d09fbfd14cc975cefabe41a/tests/ragger/test_domain_name.py#L70

Reply:
not easy to do because depending on the test we have first a warning to approve before starting the review

Reply:
Why not... 
But take a look at what is done on app-ethereum like this: https://github.com/LedgerHQ/app-ethereum/blob/76e8690a68fb84910d09fbfd14cc975cefabe41a/src_nbgl/ui_approve_tx.c#L223

The corresponding tests are: https://github.com/LedgerHQ/app-ethereum/blob/76e8690a68fb84910d09fbfd14cc975cefabe41a/tests/ragger/test_sign.py#L253

and the snapshots: https://github.com/LedgerHQ/app-ethereum/tree/develop/tests/ragger/snapshots/stax/test_sign_blind_simple